src/index.ts

```diff
@@ -27,26 +29,23 @@ export const useWallet: UseWallet = () => {
   const provider = useStore(state => state.provider);
   const web3Modal = useStore(state => state.web3Modal);
 
-  useEffect(() => {
-    useStore.setState({ web3Modal: new Web3Modal() });
-  }, []);
-
-  const connect: ConnectWallet = async opts => {
+  const connect: ConnectWallet = useCallback(async opts => {
     // Launch modal with the given options
     const web3Modal = new Web3Modal(opts);
-    useStore.setState({ web3Modal });
```

Comment:
It also wasn't clear why you'd want to perform two `setState` calls instead of a single one below. Moving this down should also reduce re-renders (though this depends on how zustand/React work together).

Reply:
Yep, good call. Hadn't considered the multiple renders.